### ISSUE TITLE
libcontainer: Do not change container state when exec'ing a process

### ISSUE THREAD

**Issue by sboeuf** [opencontainers/runc#1767]
I was trying to exec a process on a container that was not `running` from Kata Containers, and I have realized that we get an issue because of the way our agent relies on `libcontainer`. Indeed, our agent is a long lasting process that keeps `libcontainer.Container` containers and `libcontainer.Process` processes pointers in memory.
The case of the `exec` being started on a `created` container does not support this behavior. It assumes the container status will be asked from the disk everytime. That's a problem in our case and I found how to fix this, by removing [this](https://github.com/opencontainers/runc/blob/v1.0.0-rc5/libcontainer/container_linux.go#L380-L383).

Basically, this __else statement__ modifies the in-memory status of the container but never saves/stores it on disk. That's a problem when you rely exclusively on in-memory statuses like we do in case of our agent.

Now, I have opened this issue to suggest that we could remove this code part, but I might be missing a real use case for that, and I am looking for guidance here.

@crosbymichael @mrunalp @cyphar please let me know what you think about this.

**Comment by crosbymichael** [comment]
I think it makes sense but maybe we should do a refresh on the state to make sure its is written.  You will have to work on the implementation some to figureout what is the right course of action.

**Comment by sboeuf** [comment]
I am glad you're okay with that @crosbymichael , and of course I'll do the implementation :).

I am not sure about what you mean by "refresh" ? My understanding is that  we don't want this state to be written on disk because that would mean the container would be considered `running` because we did an `exec`, which would be wrong because the container workload would not have been started.
